linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/IngressIdentifier.scala

```diff
@@ -0,0 +1,73 @@
+package io.buoyant.linkerd.protocol.http
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.twitter.conversions.time._
+import com.twitter.finagle.Stack.Params
+import com.twitter.finagle.buoyant.Dst
+import com.twitter.finagle.http.Request
+import com.twitter.finagle.param.Label
+import com.twitter.finagle.service.Backoff
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.finagle.{http, Service, Dtab, Path}
+import com.twitter.util.{Duration, Future, Timer}
+import io.buoyant.config.types.Port
+import io.buoyant.k8s.{ClientConfig, IngressCache, Ns, v1beta1}
+import io.buoyant.linkerd.IdentifierInitializer
+import io.buoyant.linkerd.protocol.HttpIdentifierConfig
+import io.buoyant.router.RoutingFactory.{IdentifiedRequest, RequestIdentification, UnidentifiedRequest, _}
+
+class IngressIdentifier(
+  pfx: Path,
+  baseDtab: () => Dtab,
+  namespace: Option[String],
+  apiClient: Service[http.Request, http.Response]
+) extends Identifier[Request] {
+
+  private[this] val unidentified: RequestIdentification[Request] =
+    new UnidentifiedRequest(s"no ingress rule matches")
+
+  private[this] val ingressWithCache = IngressCache.mk(namespace, apiClient)
+
+  override def apply(req: Request): Future[RequestIdentification[Request]] = {
+    val hostHeader = req.headerMap.get("Host")
+    val matchingPaths = ingressWithCache.get(None, None).getMatchingPath(hostHeader, req.path)
+    matchingPaths.flatMap { paths =>
+      paths.headOption match {
+        case None => Future.value(unidentified)
+        case Some(a) =>
+          val path = pfx ++ Path.Utf8(a.namespace, a.port, a.svc)
+          val dst = Dst.Path(path, baseDtab(), Dtab.local)
+          req.uri = "/"
```

Comment:
I would ask if clearing the URI is part of the ingress controller spec, except that I'm guessing the ingress controller spec is not a thing that exists.  Does the golang ingress controller clear the uri like this?